server/src/test/java/io/crate/expression/operator/any/AnyEqOperatorTest.java

```diff
@@ -69,4 +76,37 @@ public void testArrayEqAnyNestedArrayMatches() {
     public void testArrayEqAnyNestedArrayDoesNotMatch() {
         assertNormalize("['foo', 'bar'] = ANY([ ['foobar'], ['foo', 'ar'], [] ])", isLiteral(false));
     }
+
+    @Test
+    public void test_uses_terms_query_for_unnested_array_refs() throws Exception {
+        QueryTester.Builder builder = new QueryTester.Builder(
+            createTempDir(),
+            THREAD_POOL,
+            clusterService,
+            Version.CURRENT,
+            "create table tbl (obj_arr array(object as (xs integer[])))"
+        );
+        List<Map<String, List<Integer>>> value1 = List.of(
+            Map.of("xs", List.of(1, 2)),
+            Map.of("xs", List.of(3, 4))
+        );
+        List<Map<String, List<Integer>>> value2 = List.of(
+            Map.of("xs", List.of(5, 6))
+        );
+        builder.indexValues("obj_arr", value1, value2);
+        try (var queryTester = builder.build()) {
+            assertThat(queryTester.runQuery("obj_arr", "1 = ANY(obj_arr['xs'])"))
+                .contains(value1);
```

Comment:
just double checking, is `contains` correct here (vs `containsExactly`)?

Reply:
It doesn't really make a difference for a single value, does it?

Reply:
Oh I didn't realize I added a second value, so it does make a difference. Too quick with the merge